hbase-server/src/main/java/org/apache/hadoop/hbase/security/access/SnapshotScannerHDFSAclController.java

```diff
@@ -290,17 +293,24 @@ public void postModifyTable(ObserverContext<MasterCoprocessorEnvironment> ctx,
   public void postDeleteNamespace(ObserverContext<MasterCoprocessorEnvironment> ctx,
       String namespace) throws IOException {
     if (checkInitialized("deleteNamespace " + namespace)) {
-      // 1. Record namespace user acl is not synced to HDFS
-      SnapshotScannerHDFSAclStorage.deleteNamespaceHdfsAcl(ctx.getEnvironment().getConnection(),
-        namespace);
-      // 2. Delete tmp namespace directory
-      /**
-       * Delete namespace tmp directory because it's created by this coprocessor when namespace is
-       * created to make namespace default acl can be inherited by tables. The namespace data
-       * directory is deleted by DeleteNamespaceProcedure, the namespace archive directory is
-       * deleted by HFileCleaner.
-       */
-      hdfsAclHelper.deleteEmptyDir(pathHelper.getTmpNsDir(namespace));
+      try (Table aclTable =
+          ctx.getEnvironment().getConnection().getTable(PermissionStorage.ACL_TABLE_NAME)) {
+        // 1. Delete namespace archive dir default ACLs
+        Set<String> users = SnapshotScannerHDFSAclStorage.getEntryUsers(aclTable,
+          PermissionStorage.toNamespaceEntry(Bytes.toBytes(namespace)));
+        hdfsAclHelper.removeNamespaceDefaultAcl(namespace, users);
```

Comment:
Is there any test case to address the  default acl removing ?   Better to have one, seems easy to miss this..

Reply:
Yes, the UT testDeleteNamespace test this case.
Before this patch, the UT just check if user can scan snapshot, after this patch, the UT also check if the namespaces directories own the default or access ACLs.